realm/src/main/java/io/realm/Realm.java

```diff
@@ -1206,7 +1206,12 @@ public void addChangeListener(RealmChangeListener listener) {
      */
     public void removeChangeListener(RealmChangeListener listener) {
         checkIfValid();
-        changeListeners.remove(new WeakReference<RealmChangeListener>(listener));
+        Iterator<WeakReference<RealmChangeListener>> iterator = changeListeners.iterator();
+        while (iterator.hasNext()) {
```

Comment:
Well... Using set doesn't solve the problem, since WeakReference doesn't override the equals method, iteration is still needed...